src/expressions.rst

```diff
@@ -2704,78 +2699,77 @@ Array and Slice Indexing Expressions
 .. rubric:: Legality Rules
 
 :dp:`fls_42ijvuqqqlvh`
-An :t:`array index expression` is an :t:`expression` that indexes into an
-:t:`array` or a :t:`slice`.
+An :t:`index expression` is an :t:`expression` that indexes into a :t:`value`
+of a :t:`type`.
 
 :dp:`fls_pc0c22asgzvw`
-An :t:`indexed array operand` is an :t:`operand` which indicates the :t:`array`
-or :t:`slice` being indexed into by an :t:`array index expression`.
+An :t:`indexed operand` is an :t:`operand` which indicates the :t:`value` of a :t:`type` implementing
+:std:`core::ops::Index` being indexed into by an :t:`index expression`.
 
 :dp:`fls_ff3sgpldn52o`
-An :t:`indexing operand` is an :t:`operand` which specifies the index of the
-:t:`array` or :t:`slice` being indexed into by an :t:`array index expression`.
+An :t:`indexing operand` is an :t:`operand` which specifies the index for the :t:`indexed operand`
+being indexed into by an :t:`index expression`.
```

Comment:
Let's relax this to

An indexing operand is an operand which specifies the index of an index expression.